openrouteservice/src/main/java/org/heigit/ors/routing/RoutingProfile.java

```diff
@@ -176,7 +176,9 @@ public static ORSGraphHopper initGraphHopper(String osmFile, RouteProfileConfigu
         // so the caching for elevation data will/can be reused across different vehicles. [the loadCntx is a single
         // Object that will shared across the (potential) multiple running instances]
         if(loadCntx.getElevationProvider() != null) {
-            gh.setElevationProvider(loadCntx.getElevationProvider());
+            if (args.has("graph.elevation.provider")) {
```

Comment:
What happens if `(loadCntx.getElevationProvider() != null) && !args.has("graph.elevation.provider")`

Reply:
the loadcontext is used to store elevationprovider if it has been instantiated before, as far as i understand to make enable caching of data, and the problem was that every later profile was assigned the provider regardless if it had elevation parameter set to true. in the case in question the next line is supposed to not run.